third-party/realdds/src/dds-defines.cpp

```diff
@@ -0,0 +1,86 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-defines.h>
+#include <realdds/dds-utilities.h>
+
+
+namespace realdds {
+
+
+
+nlohmann::json video_intrinsics::to_json() const
+{
+    return nlohmann::json::array( {
+        width, height, principal_point_x, principal_point_y, focal_lenght_x, focal_lenght_y, distortion_model,
+        distortion_coeffs[0], distortion_coeffs[1], distortion_coeffs[2], distortion_coeffs[3], distortion_coeffs[4]
+    } );
+}
+
+/* static  */ video_intrinsics video_intrinsics::from_json( nlohmann::json const & j )
+{
+    video_intrinsics ret;
+    int index = 0;
+
+    ret.width = utilities::json::get< int >( j, index++ );
```

Comment:
The only issue here is when you add a new value to the struct (in the middle)
and you forget to update here, you ruin all other fields..
Is it worth it?

Reply:
How about monitoring it from a sniffer?
The user will do "ros2 topic echo" and see just values with no context, right?
I think since the topic is small (intrinsics for every stream type right?) we may want to be more conservative ..

Your call..

Reply:
I will check with Yoram's team about their limitations, currently we tried to be under 4K bytes.
Alternatively we could split a stream header to several messages but I like it less.

Also, remember that when using an array, the values are kept in order of placement. When using dictionary they are sorted in lexicographic key order

Reply:
Intrinsics struct size is small no?
Is it part of a bigger struct?

Reply:
12 fields (video) or 18 float (motion) fields. When sending as text (JSON) most are 16 characters long (some more) and we have a set for each resolution supported by the stream, so not that small...

D435i color stream intrinsics array for example:[[320,180,161.7417755126953,90.47455596923828,227.0221710205078,227.1049346923828,2,0.0,0.0,0.0,0.0,0.0],[320,240,162.32237243652344,120.63274383544922,302.69622802734375,302.80657958984375,2,0.0,0.0,0.0,0.0,0.0],[424,240,214.32235717773438,120.63274383544922,302.69622802734375,302.80657958984375,2,0.0,0.0,0.0,0.0,0.0],[640,360,323.4835510253906,180.94911193847656,454.0443420410156,454.2098693847656,2,0.0,0.0,0.0,0.0,0.0],[640,480,324.6447448730469,241.26548767089844,605.3924560546875,605.6131591796875,2,0.0,0.0,0.0,0.0,0.0],[848,480,428.64471435546875,241.26548767089844,605.3924560546875,605.6131591796875,2,0.0,0.0,0.0,0.0,0.0],[960,540,485.2253112792969,271.4236755371094,681.0665283203125,681.3148193359375,2,0.0,0.0,0.0,0.0,0.0],[1280,720,646.9671020507813,361.8982238769531,908.0886840820313,908.4197387695313,2,0.0,0.0,0.0,0.0,0.0],[1920,1080,970.4506225585938,542.8473510742188,1362.133056640625,1362.629638671875,2,0.0,0.0,0.0,0.0,0.0]],

Reply:
With field names it will be about twice the size

Reply:
Got it..